homeassistant/components/kef/media_player.py

```diff
@@ -99,9 +100,16 @@
         sources,
     )
 
-    latitude = config.data[CONF_LATITUDE]
-    longitude = config.data[CONF_LONGITUDE]
-    unique_id = f"kef-{latitude}-{longitude}"
+    try:
+        if ipaddress.ip_address(host).version == 6:
+            mode = "ip6"
+        else:
+            mode = "ip"
+    except ValueError:
+        mode = "hostname"
+    mac = await hass.async_add_executor_job(partial(get_mac_address, **{mode: host}))
+
+    unique_id = f"kef-{mac}"
```

Comment:
We need to guard for return value `None` for `mac`. In that case we should return `None` as `unique_id`.

Reply:
Fixed in c6df8a0d453617e450da4fe266f02e4e43dabc37.